PROJECT-CLEARWATER-MIB

```diff
@@ -2447,4 +2450,237 @@ mementoGroup OBJECT-GROUP
+    SYNTAX      AstaireBandwidthTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization bandwidth."
+    INDEX       { astaireBandwidthScope }
+    ::= { astaireBandwidthTable 1 }
+
+AstaireBandwidthTableEntry ::= SEQUENCE
+{
+    astaireBandwidthScope          INTEGER,
+    astaireBandwidth               Unsigned32
+}
+
+astaireBandwidthScope OBJECT-TYPE
+    SYNTAX      INTEGER {
+                  scopeCurrent5SecondPeriod(1)
+                }
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "The period over which the statistics were collected."
+    ::= { astaireBandwidthTableEntry 1 }
+
+astaireBandwidth OBJECT-TYPE
+    SYNTAX      Unsigned32
+    MAX-ACCESS  read-only
+    STATUS      current
+    DESCRIPTION "Resynchronization bandwidth (bytes per second) over the period."
+    ::= { astaireBandwidthTableEntry 2 }
+
+astaireConnectionTable OBJECT-TYPE
+    SYNTAX SEQUENCE OF AstaireConnectionTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics on a per-connection basis."
+    ::= { astaire 6 }
+
+astaireConnectionTableEntry OBJECT-TYPE
+    SYNTAX      AstaireConnectionTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics for a specific connection."
+    INDEX { astaireConnectionInetAddrType,
+            astaireConnectionInetAddr      }
+    ::= { astaireConnectionTable 1 }
+
+AstaireConnectionTableEntry ::= SEQUENCE
+{
+  astaireConnectionInetAddrType           InetAddressType,
+  astaireConnectionInetAddr               InetAddress,
+  astaireConnectionBucketsNeedingResync   Unsigned32,
+  astaireConnectionBucketsResynchronized  Unsigned32,
+  astaireConnectionBucketTable            SEQUENCE OF astaireConnectionBucketTableEntry
+}
+
+astaireConnectionInetAddrType OBJECT-TYPE
+    SYNTAX      InetAddressType
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Inet address type of the connection peer"
+    ::= { astaireConnectionTableEntry 1 }
+
+astaireConnectionInetAddr OBJECT-TYPE
+    SYNTAX      InetAddress (SIZE (1..16))
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Inet address of the connection peer"
+    ::= { astaireConnectionTableEntry 2 }
+
+astaireConnectionBucketsNeedingResync OBJECT-TYPE
+    SYNTAX      Unsigned32
+    MAX-ACCESS  read-only
+    STATUS      current
+    DESCRIPTION "Count of buckets needing to be synchronized over this
+                 connection for the current resynchronization operation."
+    ::= { astaireConnectionTableEntry 3 }
+
+astaireConnectionBucketsResynchronized OBJECT-TYPE
+    SYNTAX      Unsigned32
+    MAX-ACCESS  read-only
+    STATUS      current
+    DESCRIPTION "Number of buckets already resynchronized over this connection
+                 during the current resynchronization operation."
+    ::= { astaireConnectionTableEntry 4 }

```

Comment:
Since we don't know when a bucket finishes resynch, this will always either equal the previous stat or 0.  I think we're ok with this, but I'm just checking.